### PR TITLE
feat: allow to deploy without a `ClusterRole`

### PR DESCRIPTION
**Pull request by eliecharra**
- [x] release `v0.0.1-alpha3` before merge

### CODE REVIEW

File: spacelift-workerpool-controller/templates/manager-rbac.yaml [hunk 1]
```diff
@@ -101,4 +103,115 @@ roleRef:
...
 - kind: ServiceAccount
   name: '{{ include "spacelift-workerpool-controller.fullname" . }}-controller-manager'
-  namespace: '{{ .Release.Namespace }}'
\ No newline at end of file
+  namespace: '{{ .Release.Namespace }}'
+{{ else }}
+{{ range $index, $namespace := .Values.controllerManager.namespaces }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
```
Comment (eliecharra) [inline spacelift-workerpool-controller/templates/manager-rbac.yaml:111]: Basically a copy paste of the `ClusterRole` with an additional `metadata.namespace` directive.

Comment (adamconnelly) [reply spacelift-workerpool-controller/templates/manager-rbac.yaml:111]: Yeah - that makes sense if you can do that. Makes it much less likely we adjust permissions and break one of the roles.

File: spacelift-workerpool-controller/values.yaml [hunk 2]
```diff
@@ -1,4 +1,8 @@
 controllerManager:
+  # If you do not specify a list of namespaces here, the controlled will be assigned a ClusterRole.
```
Comment (adamconnelly) [inline spacelift-workerpool-controller/values.yaml:2]:
```suggestion
  # If you do not specify a list of namespaces here, the controller will be assigned a ClusterRole, and will be able to manage WorkerPools across all namespaces in your cluster.
```

**Review by adamconnelly — APPROVED** [review]
Looks good. Just made some suggestions about tweaking some comments, and I agree about templating the permissions if possible.

**Review by adamconnelly — APPROVED** [review]

**Pull request merged**